lib/Parser.js

```diff
@@ -1454,9 +1470,59 @@ class Parser extends Tapable {
 			free
 		};
 	}
-
 }
 
+const originalPlugin = Parser.prototype.plugin;
+Parser.prototype.plugin = function pluginIntoParser(inHook, potentialDef, opts) {
```

Comment:
This is super gross, but basically is allowing us to serialize parser plugins...

This api would be a lot simpler if we didn't make `Parser` extend `Tapable`. We could eliminate the public `plugin` and `apply` methods, then expose an `apply` that takes a plugin definition like...
```
{
  path: '/path/to/plugin',
  options: ['options', 'to', 'initialize', 'with']
}
```

We could then revert most of the ParserPlugin changes and the ParserHelper changes.